File: tests/fixtures/python-fastapi/src/gateways/payment.py
import stripe

# @lattice:boundary stripe
def charge(amount: float, token: str) -> dict:
    payload = build_stripe_payload(amount, token)
    return stripe.charges.create(**payload)

def build_stripe_payload(amount, token):
    return {"amount": int(amount * 100), "source": token}

File: tests/fixtures/typescript-express/src/gateways/payment.ts
import Stripe from "stripe";

const stripe = new Stripe("sk_test");

// @lattice:boundary stripe
export async function charge(data: unknown): Promise<ChargeResult> {
	const payload = buildPayload(data);
	return stripe.charges.create(payload);
}

function buildPayload(_data: unknown): Stripe.ChargeCreateParams {
	return { amount: 100, currency: "usd" };
}

interface ChargeResult {
	id: string;
	status: string;
}
